source/common/http/conn_pool_grid.h

```diff
@@ -19,6 +19,11 @@ class ConnectivityGrid : public ConnectionPool::Instance,
     std::vector<Http::Protocol> protocols_;
   };
 
+  enum class StreamCreationResult {
```

Comment:
I'm slightly surprised this doesn't exist at the general connection pool layer. But this is super great in any case!